main/main.go

```diff
@@ -110,12 +110,14 @@ func main() {
 	// set up graceful shutdown handling
 	go shutdown(cancel)
 
-	// set up synchronous chaining routine
-	chainingJobs := make(chan ChainingRequest, conf.RequestBufSize)
+	// set up synchronous chaining routines for each identity
+	chainingJobs := make(map[string]chan ChainingRequest, len(conf.Devices))
 
-	g.Go(func() error {
-		return s.chainer(chainingJobs)
-	})
+	for id := range conf.Devices {
+		jobs := make(chan ChainingRequest, conf.RequestBufSize)
+		chainingJobs[id] = jobs
+		s.startChainer(g, id, jobs)
+	}
```

Comment:
I think this is a good solution, but will not allow us to add/remove devices at runtime. Will try to make an issue about this tomorrow.

Reply:
i was thinking we could just call `s.startChainer(g, id, jobs)` with a newly created channel from a device-adding function and add the new channel to the `chainingJobs` map.  For that we will need a getter/setter with mutex for the `chainingJobs` map, but other than that I saw no issue. Did I miss something?

Reply:
for removing devices, a function could simply close the channel and remove it from the map